### PR TITLE
[DCA] Use Role instead of ClusterRole for secrets RBAC

### PR DESCRIPTION
**Pull request by ahmed-mez**
#### What this PR does / why we need it:

Apply this change https://github.com/DataDog/datadog-agent/pull/7754

#### Special notes for your reviewer:

DCA 1.12 must be released before merging this PR

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/templates/cluster-agent-rbac.yaml [hunk 1]
```diff
@@ -314,6 +311,42 @@ metadata:
...
+apiVersion: {{ template "rbac.apiVersion" . }}
+kind: RoleBinding
+metadata:
+  labels:
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    app.kubernetes.io/name: "{{ template "datadog.fullname" . }}"
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+  name: "{{ template "datadog.fullname" . }}-cluster-agent-rolebinding"
```
Comment (L3n41c) [inline charts/datadog/templates/cluster-agent-rbac.yaml:340]: The `ClusterRoleBinding` is just named `{{ template "datadog.fullname" . }}-cluster-agent`. I.e. the name of the object is not suffixed by its type.
As this `RoleBinding` is a different kind, we shouldn’t have any name conflict.

So, we could use:
```suggestion
  name: "{{ template "datadog.fullname" . }}-cluster-agent"
```

**Review by L3n41c — APPROVED** [review]
Just left a small remark about the fact that we usually not suffix object names with their kind.

File: charts/datadog/templates/cluster-agent-rbac.yaml (hunk 1 above)
Comment (ahmed-mez) [reply charts/datadog/templates/cluster-agent-rbac.yaml:340]: we actually have a rolebinding with this name https://github.com/DataDog/helm-charts/blob/master/charts/datadog/templates/cluster-agent-rbac.yaml#L353

**Review by L3n41c — APPROVED** [review]
👍🏻

**Pull request merged**